api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -79,13 +79,14 @@ public static PreparedStatement prepareTableSelectStatement(Connection c,
                                                                 String storageKey,
                                                                 String[] fields,
                                                                 String[] values,
-                                                                Persistable p) {
+                                                                Persistable p) throws IllegalArgumentException{
```

Comment:
Adding to javadoc is definitely the primary utility. @wpride can you make sure to add the circumstances under which this gets thrown as a field in the method level javadoc comment? I'm fine with following phillip's pattern of not adding to the signature if there's a doc comment